mm2src/coins/lp_coins.rs

```diff
@@ -5599,3 +5603,74 @@ pub mod for_tests {
         }
     }
 }
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20ApproveRequest {
+    coin: String,
+    spender: EthAddress,
+    amount: BigDecimal,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Erc20AllowanceRequest {
+    coin: String,
+    spender: EthAddress,
+}
+
+#[derive(Debug, Deserialize, Display, EnumFromStringify, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum Erc20CallError {
+    #[display(fmt = "No such coin {}", coin)]
+    NoSuchCoin { coin: String },
+    #[display(fmt = "Coin not supported {}", coin)]
+    CoinNotSupported { coin: String },
+    #[from_stringify("NumConversError")]
+    #[display(fmt = "Invalid param: {}", _0)]
+    InvalidParam(String),
+    #[from_stringify("TransactionErr")]
+    #[display(fmt = "Transaction error {}", _0)]
+    TransactionError(String),
+    #[from_stringify("Web3RpcError")]
+    #[display(fmt = "Web3 RPC error {}", _0)]
+    Web3RpcError(String),
+}
+
+impl HttpStatusCode for Erc20CallError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            Erc20CallError::NoSuchCoin { .. }
+            | Erc20CallError::CoinNotSupported { .. }
+            | Erc20CallError::InvalidParam(_)
+            | Erc20CallError::TransactionError(_)
+            | Erc20CallError::Web3RpcError(_) => StatusCode::BAD_REQUEST,
+        }
+    }
+}
+
+/// Call allowance method for ERC20 tokens (see https://eips.ethereum.org/EIPS/eip-20#approve).
+/// Returns BigDecimal allowance value.
+pub async fn allowance_rpc(ctx: MmArc, req: Erc20AllowanceRequest) -> MmResult<BigDecimal, Erc20CallError> {
+    let eth_coin = find_erc20_eth_coin(&ctx, &req.coin).await?;
+    let wei = eth_coin.allowance(req.spender).compat().await?;
+    let amount = u256_to_big_decimal(wei, eth_coin.decimals())?;
+    Ok(amount)
+}
+
+/// Call approve method for ERC20 coins (see https://eips.ethereum.org/EIPS/eip-20#allowance).
+/// Returns approval transaction hash.
+pub async fn approve_rpc(ctx: MmArc, req: Erc20ApproveRequest) -> MmResult<BytesJson, Erc20CallError> {
+    let eth_coin = find_erc20_eth_coin(&ctx, &req.coin).await?;
+    let amount = wei_from_big_decimal(&req.amount, eth_coin.decimals())?;
+    let tx = eth_coin.approve(req.spender, amount).compat().await?;
+    Ok(tx.tx_hash_as_bytes())
```

Comment:
this would return a result like that 
```
{
  "mmrpc": "2.0",
  "result": "5136701f11060010841c9708c3eb26f6606a070b8ae43f4b98b6d7b10a545258",
  "id": null
}
```

Reply:
It should include the `0x` prefix

Reply:
Not fixed yet.

Reply:
fixed in 65c4853fd1c34ef8c3527a9becc31e04b12dcd38
I can see couple places where 0x is not added though: [send_raw_tx in eth](https://github.com/KomodoPlatform/komodo-defi-framework/blob/35e92394928825c337f246f8e19fbfab1f65c4a8/mm2src/coins/eth.rs#L2203) and  [TransactionNftDetails](https://github.com/KomodoPlatform/komodo-defi-framework/blob/35e92394928825c337f246f8e19fbfab1f65c4a8/mm2src/coins/eth.rs#L975)
We apparently need a dedicated to_string conversion fn for tx_hash.

(But is it convenient for GUI when we return tx hash in different formats for different coins?)

Reply:
> But is it convenient for GUI when we return tx hash in different formats for different coins

No, we should fix those cases. tx hashes and addresses should always be displayed/sent to GUIs with `0x` prefix